test/common.py

```diff
@@ -359,4 +360,118 @@ def setdata(self,data):
+                        0,  0, 0,  0, 0, 0, 0, 0,
+                        0,  0, 0,  0, 1, 0, 0, 0
+                       ]
+
+
+        self.red_pixels_2 = [
+                        0, 12, 20,  0,  0,   2,   3,  4,
+                        0, 0,  2,  3,  4,   5,   1,  0,
+                        0,  0,  0,  0,  0,   0,  90,  0,
+                        0,  0,  0, 10,  0,   0,   0,  9,
+                        1,  1,  1,  1,  2,   3,   4,  5,
+                        1,  2,  3,  4,  1,   2,   0,  0,
+                        0,  0,  0,  0,  0,   0,   0,  0,
+                        0,  0,  0,  0,  0,   0,   0,  0
+                       ]
+
+        self.green_pixels_1 = [
+                          11, 12, 0,  0, 0, 0, 0, 0,
+                           0,  0, 0,  0, 0, 0, 0, 0,
+                           0,  0, 0, 10, 2, 3, 4, 0,
+                           0,  0, 0,  0, 1, 0, 0, 0,
+                           0,  0, 1,  1, 2, 3, 4, 5,
+                           1,  2, 3,  4, 1, 2, 0, 0,
+                           0,  0, 0,  0, 0, 0, 0, 0,
+                           0,  0, 0,  0, 1, 0, 0, 0
+                         ]
+
+        self.green_pixels_2 = [
+                          13, 12, 20,  0,  0,   0,   0,  0,
+                           0,  0,  0,  0,  0,   0,   0,  0,
+                           0,  0,  0,  0,  0,   0,   0,  0,
+                           0,  0,  0,  0,  0,   0,   0,  0,
+                           0,  0,  0,  0,  0,   0,   0,  0,
+                           0,  0,  0,  0,  0,   0,   0,  0,
+                           0,  0,  0,  0,  0,   0,   0,  1,
+                           1,  0,  0,  0,  1,  32,   4,  2
+                          ]
+
+        self.blue_pixels_1 = [
+                         11, 12, 0,  0, 0, 0, 0, 0,
+                          0,  0, 0,  0, 0, 0, 0, 0,
+                          0,  0, 0,  0, 0, 0, 0, 0,
+                          0,  0, 0,  0, 0, 0, 0, 0,
+                          0,  0, 0,  1, 2, 3, 4, 5,
+                          1,  2, 3,  4, 1, 2, 0, 0,
+                          0,  0, 0,  0, 0, 0, 0, 0,
+                          0,  0, 0,  0, 0, 0, 0, 1
+                        ]
+
+        self.blue_pixels_2 = [
+                         16, 12, 20,  0,  0,   2,   3,  4,
+                          0,  0,  2,  3,  4,   5,   1,  0,
+                          0,  0,  0,  0,  0,   0,  90,  0,
+                          0,  0,  0, 10,  0,   0,   0,  9,
+                          1,  1,  1,  1,  2,   3,   4,  5,
+                          1,  2,  3,  4,  1,   2,   0,  1,
+                          1,  0,  0,  0,  0,   0,   0,  1,
+                          1,  0,  0,  0,  1,  32,   4,  2
+                        ]
+        # Get the output from the reference module.
+        self.output_red_pixels_1 = rle_code(self.red_pixels_1);
+
+    def setdata(self,data):
+        self.data = data
+        for i in range(64):
+            temp=self.data[i]
```

Comment:
The data we get contains amplitude and run length as a single number. In order to extract the values we use mod 4096 to get the last 12bits and then shift to extract the next 4 representing the run length.

Reply:
DONE